app/services/fetch_coronavirus_statistics_service.rb

```diff
@@ -135,13 +137,33 @@ def weekly_tests(latest_tests, data)
       weeks_with_tests << day_cases["newPositiveTests"] if latest_tests["date"] >= day_cases["date"]
     end
 
-    current_week_tests = weeks_with_tests.first(7).compact
-    return {} unless current_week_tests.size == 7
+    total_current_week_tests = current_week_positive_tests(weeks_with_tests)
+    total_previous_week_tests = previous_week_positive_tests(weeks_with_tests)
 
-    total_current_week_tests = current_week_tests.inject(:+)
+    return {} unless total_current_week_tests && total_previous_week_tests
 
     {
       current_week_positive_tests: total_current_week_tests,
+      current_week_positive_tests_change_number: total_current_week_tests - total_previous_week_tests,
+      current_week_positive_tests_change_percentage: percentage_change(total_current_week_tests, total_previous_week_tests),
     }
   end
+
+  def current_week_positive_tests(weeks_with_tests)
+    current_week_tests = weeks_with_tests.first(7).compact
+    return unless current_week_tests.size == 7
+
+    current_week_tests.inject(:+)
+  end
+
+  def previous_week_positive_tests(weeks_with_tests)
+    previous_week_cases = weeks_with_tests[7..13].compact
+    return unless previous_week_cases.size == 7
+
+    previous_week_cases.inject(:+)
+  end
+
+  def percentage_change(current_total, previous_total)
```

Comment:
Maths is good, nice one :) 
Amazed there's not a gem or rails library for this though! (I checked)

Reply:
So did I!

Reply:
Might write a gem as a side project, open source community and all that (love off topic review comments that send emails (sorry for the spam (also love brackets in brackets (i hope I don't embarrass myself by not closing them all))))